### PR TITLE
Add top-level clean make target

### PR DESCRIPTION
**Pull request by adrianreber**
And propagate clean to all directories.

### CODE REVIEW

File: test/loop/Makefile [hunk 1]
```diff
@@ -2,3 +2,8 @@ CC ?= gcc
 
 loop: loop.c
 	$(CC) $^ -o $@
+
+clean:
+	rm -f loop
```
Comment (snprajwal) [inline test/loop/Makefile:7]: I think we would want to silence the command itself, like in the other Makefiles
```suggestion
	@rm -f loop
```

File: Makefile [hunk 2]
```diff
@@ -38,4 +38,8 @@ vendor:
 	$(GO) mod vendor
 	$(GO) mod verify
 
-.PHONY: build test lint vendor coverage codecov
+clean:
+	make -C crit/ clean
+	make -C test/ clean
```
Comment (rst0git) [inline Makefile:43]: we should use `$(MAKE)` for [recursive invocations](https://www.gnu.org/software/make/manual/html_node/Recursion.html): https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable

**Comment by adrianreber** [comment]
Applied changes as suggested.

**Review by rst0git — APPROVED** [review]
LGTM

**Pull request merged**